consensus/consortium/v2/consortium.go

```diff
@@ -408,6 +416,53 @@ func (c *Consortium) verifyCascadingFields(chain consensus.ChainHeaderReader, he
 	return c.verifySeal(chain, header, parents, snap)
 }
 
+// snapshotAtConsortiumFork is expected to have to the same
+// behavior as snapshot. However, the validator list is read from
+// header instead of statedb to support snap sync. We try to keep
+// the snapshot function unchanged to minimize the changing effect.
+func (c *Consortium) snapshotAtConsortiumFork(
+	chain consensus.ChainHeaderReader,
+	number uint64,
+	hash common.Hash,
+	forkedHeader *types.Header,
+	parents []*types.Header,
+) (*Snapshot, error) {
+	var validators []common.Address
+
+	snap, err := loadSnapshot(c.config, c.signatures, c.db, hash, c.ethAPI, c.chainConfig)
+	if err == nil {
+		log.Trace("Loaded snapshot from disk", "number", number, "hash", hash.Hex())
+		return snap, nil
+	}
+
+	extraData, err := finality.DecodeExtra(forkedHeader.Extra, false)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, validator := range extraData.CheckpointValidators {
+		validators = append(validators, validator.Address)
+	}
+
+	snap = newSnapshot(c.chainConfig, c.config, c.signatures, number, hash, validators, nil, c.ethAPI)
+
+	// load v1 recent list to prevent recent producing-block-validators produce block again
+	snapV1 := c.v1.GetSnapshot(chain, number, hash, parents)
```

Comment:
https://github.com/minh-bq/ronin/blob/0a699bc486bfeae669feaa3ec8e35b431206229d/consensus/consortium/v1/consortium.go#L711 seem the error of GetSnapshot is swallowed here ? If any error occurs here, we only receive message snapshot V1 is not available ? Could be return err from GetSnapshot function and return here ?

Reply:
This code is executed only once at the Consortium block which is in the past, so I don't think adding error here is necessary anymore.